pkg/protocol/engine/filter/blockfilter/filter.go

```diff
@@ -68,102 +67,95 @@ func New(apiProvider api.Provider, opts ...options.Option[Filter]) *Filter {
 // ProcessReceivedBlock processes block from the given source.
 func (f *Filter) ProcessReceivedBlock(block *model.Block, source network.PeerID) {
 	blockSlotAPI := f.apiProvider.APIForSlot(block.ID().Index())
-	protocolParams := blockSlotAPI.ProtocolParameters()
 
 	if blockSlotAPI.Version() != block.ProtocolBlock().ProtocolVersion {
-		f.events.BlockFiltered.Trigger(&filter.BlockFilteredEvent{
+		f.events.BlockPreFiltered.Trigger(&filter.BlockPreFilteredEvent{
 			Block:  block,
 			Reason: ierrors.Wrapf(ErrInvalidBlockVersion, "invalid protocol version %d (expected %d) for epoch %d", block.ProtocolBlock().ProtocolVersion, blockSlotAPI.Version(), blockSlotAPI.TimeProvider().EpochFromSlot(block.ID().Index())),
 			Source: source,
 		})
 
 		return
 	}
-
-	// Check if the block is trying to commit to a slot that is not yet committable.
-	// This check, together with the optsMaxAllowedWallClockDrift makes sure that no one can issue blocks with commitments in the future.
-	if block.ProtocolBlock().SlotCommitmentID.Index() > 0 && block.ProtocolBlock().SlotCommitmentID.Index()+protocolParams.EvictionAge() > block.ID().Index() {
-		f.events.BlockFiltered.Trigger(&filter.BlockFilteredEvent{
+	// Verify the timestamp is not too far in the future.
+	timeDelta := time.Since(block.ProtocolBlock().IssuingTime)
+	if timeDelta < -f.optsMaxAllowedWallClockDrift {
+		f.events.BlockPreFiltered.Trigger(&filter.BlockPreFilteredEvent{
 			Block:  block,
-			Reason: ierrors.Wrapf(ErrCommitmentNotCommittable, "block at slot %d committing to slot %d", block.ID().Index(), block.ProtocolBlock().SlotCommitmentID.Index()),
+			Reason: ierrors.Wrapf(ErrBlockTimeTooFarAheadInFuture, "issuing time ahead %s vs %s allowed", -timeDelta, f.optsMaxAllowedWallClockDrift),
 			Source: source,
 		})
 
 		return
 	}
 
-	// Verify the timestamp is not too far in the future.
-	timeDelta := time.Since(block.ProtocolBlock().IssuingTime)
-	if timeDelta < -f.optsMaxAllowedWallClockDrift {
-		f.events.BlockFiltered.Trigger(&filter.BlockFilteredEvent{
+	minCommittableAge := blockSlotAPI.ProtocolParameters().MinCommittableAge()
+	maxCommittableAge := blockSlotAPI.ProtocolParameters().MaxCommittableAge()
+
+	// check that commitment is within allowed range.
+	if minCommittableAge > 0 &&
+		block.ProtocolBlock().SlotCommitmentID.Index() > 0 &&
```

Comment:
I find this a bit hard to read. Can we use `commitmentIndex` here and also add comments for the different conditions, as well as some extra brackets for the AND conditions?